python-checks/src/main/java/org/sonar/python/checks/regex/OctalEscapeCheck.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.regex;
+
+import java.util.Collections;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.ast.CharacterTree;
+import org.sonarsource.analyzer.commons.regex.ast.RegexBaseVisitor;
+
+@Rule(key = "S6537")
+public class OctalEscapeCheck extends AbstractRegexCheck {
+
+  public static final String MESSAGE = "Replace with hexadecimal code";
```

Comment:
I'd suggest `Consider replacing this octal escape sequence with a Unicode or hexadecimal sequence instead.`, or at least something that doesn't just suggest to replace with hexadecimal code (mainly because it's not the only fix, but also because it would be unclear for users if they actually meant to have a backreference).